### DISCUSSION TITLE
Query or mutation?

### DISCUSSION THREAD

**Discussion by andrewslock** [TanStack/query#1546] [Q&A] [answered]
I've got a question about React Query ideology

As far as I know, queries are used to fetch data from the server. Usually you use it for GET requests. Mutations, on the contrary, are used to modify data on the server. You use those for anything but GET requests.

In my case I have a POST request I use to send my data from web page to a printer. The thing is, this request doesn't modify any data on the server side, neither does it fetch any data for me. So I am not sure whether I should use mutation or query.

Additionally, I'd like this request to be made on the first render of my component. Thus, if using mutation, I have to call it in a useEffect hook, which seems to be too much

So what should I do?

**Comment by TkDodo** [comment, marked as answer]
I think useMutation in an effect seems the right choice. With useQuery, you should be ready that this fires quite often - for example, when the user loses their internet connection and then reconnects. If your query performs effects on the server, this is not a good idea. Also, you’ve said that your post request doesn’t return anything, so i assume it stores something on the printer? Or even prints a page :)
Anyways, those are all side effects, so useMutation sounds good to me.

**Reply by andrewslock** [reply]
Ok, thanks. But is there any way for a mutation to be made on its own (like a query) (without calling mutate/mutateAsync)? Because I really dont feel like using useEffect

**Reply by TkDodo** [reply]
There is not, and that’s on purpose. Since mutations perform a side effect on the server, you have to explicitly invoke them. You can abstract that away in a `useMountMutation` or so custom hook that is just `useMutation` + `useEffect` :)

**Comment by tannerlinsley** [comment]
If it isn't fetching data or "subscribing" your app to some remote data (`useQuery`), then it's either a mutation or general RPC.

Use `useMutation` 👍
